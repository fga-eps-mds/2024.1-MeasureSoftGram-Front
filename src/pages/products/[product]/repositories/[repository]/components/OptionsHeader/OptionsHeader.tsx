import { Box, Button } from '@mui/material';

import React from 'react';

interface Props {
  title: string;
  isHistoricOpen: boolean;
  setIsHistoricOpen: (arg: boolean) => void;
}

function OptionsHeader({ title, isHistoricOpen, setIsHistoricOpen }: Props) {
  return (
    <Box
      display="flex"
      flexDirection="row"
      height={60}
      alignItems="center"
      justifyContent="space-between"
      gap="2rem"
    >
      <h2 style={{ color: '#113D4C', fontWeight: '500', fontSize: '25px' }}>{title}</h2>
      <Box
        style={{
          width: '50%'
        }}
        display="flex"
        flexDirection="row"
        justifyContent="flex-end"
        gap="1rem"
      >
        <Button
          onClick={() => setIsHistoricOpen(true)}
          variant={isHistoricOpen ? 'contained' : 'outlined'}
        >
          Histórico
        </Button>
        <Button
          onClick={() => setIsHistoricOpen(false)}
          variant={!isHistoricOpen ? 'contained' : 'outlined'}
        >
          Cenário Atual
        </Button>
      </Box>
    </Box >
  );
}

export default OptionsHeader;
