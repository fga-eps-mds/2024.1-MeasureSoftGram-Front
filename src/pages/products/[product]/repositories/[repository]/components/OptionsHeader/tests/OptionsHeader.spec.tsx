import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import OptionsHeader from '../OptionsHeader';
import '@testing-library/jest-dom';
import { useTranslation } from 'react-i18next';


describe('OptionsHeader', () => {

  it('should render the title', () => {
    const isOpen = true;
    const { getByText } = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={isOpen}
        setIsHistoricOpen={() => { }}
      />
    );
    expect(getByText('Teste')).toBeTruthy();
  });

  it('should call setIsHistoricOpen(true) when "Histórico" is clicked', () => {
    const setIsHistoricOpen = jest.fn();
    const { getByText } = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={false}
        setIsHistoricOpen={setIsHistoricOpen}
      />
    );

    const historicButton = getByText(t('history'));
    fireEvent.click(historicButton);
    expect(setIsHistoricOpen).toHaveBeenCalledWith(true);
  });

  it('should call setIsHistoricOpen(false) when "Cenário Atual" is clicked', () => {
    const setIsHistoricOpen = jest.fn();
    const { getByText } = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={false}
        setIsHistoricOpen={setIsHistoricOpen}
      />
    );
    const currentButton = getByText('Cenário Atual');
    fireEvent.click(currentButton);
    expect(setIsHistoricOpen).toHaveBeenCalledWith(false);
  });

  it('"Histórico" button should render with class "contained" when isHistoricOpen is true', () => {
    const isOpen = true;
    const { getByText } = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={isOpen}
        setIsHistoricOpen={() => { }}
      />
    );
    const statusButton = getByText('Histórico');
    expect(statusButton).toHaveClass('MuiButton-contained');
  });

  it('"Cenário Atual" button should render with class "contained" when isHistoricOpen is false', () => {
    const { getByText } = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={false}
        setIsHistoricOpen={() => { }}
      />
    );
    const statusButton = getByText('Cenário Atual');
    expect(statusButton).toHaveClass('MuiButton-contained');
  });

  it('should match snapshot', () => {
    const tree = render(
      <OptionsHeader
        title="Teste"
        isHistoricOpen={false}
        setIsHistoricOpen={() => { }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
