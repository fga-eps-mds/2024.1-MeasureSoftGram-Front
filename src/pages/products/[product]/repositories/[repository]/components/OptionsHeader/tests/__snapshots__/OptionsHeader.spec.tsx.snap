// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsHeader should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1ntb6q6"
      >
        <h2
          style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
        >
          Teste
        </h2>
        <div
          class="MuiBox-root css-1vr4rvm"
          style="width: 50%;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            data-testid="history-button"
            tabindex="0"
            type="button"
          >
            history
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-current-scenario"
            tabindex="0"
            type="button"
          >
            current-scenario
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1ntb6q6"
    >
      <h2
        style="color: rgb(17, 61, 76); font-weight: 500; font-size: 25px;"
      >
        Teste
      </h2>
      <div
        class="MuiBox-root css-1vr4rvm"
        style="width: 50%;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          data-testid="history-button"
          tabindex="0"
          type="button"
        >
          history
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="button-current-scenario"
          tabindex="0"
          type="button"
        >
          current-scenario
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
